README.md

```diff
@@ -0,0 +1,169 @@
+# V-Server Setup Documentation
+
+## Introduction
+
+This document outlines the steps that were followed to set up a V-Server for project deployment. It covered configuring SSH access, installing and configuring a web server (Nginx), and setting up GitHub integration.
+
+---
+
+## SSH Configuration
+
+### 1. Generated SSH Key Pair
+
+To securely connect to the server, an SSH key pair was generated.
+
+On the local machine, the following command was run to generate the SSH key pair:
+
+```bash
+ssh-keygen -t ed25519
+
+```
+
+This generated an SSH key pair (public and private keys) in the ~/.ssh/ directory.
+
+### 2. Copied the Public Key to the Server
+After generating the SSH key pair, the public key was copied to the server. This allowed authentication using the SSH key rather than a password.
+
+The ssh-copy-id command was used to transfer the public key:
+```bash
+ssh-copy-id -i /Users/abdul/.ssh/id_ed25519.pub aalibrahim@128.140.8.32
+```
+### 3. Disabled Password Authentication
+To enhance security, password-based login was disabled, and only SSH key authentication was allowed.
+
+The SSH configuration file was edited as follows:
+
+```bash
+sudo nano /etc/ssh/sshd_config
+```
+
+The following line was modified:
+
+```bash
+PasswordAuthentication no
+```
+The changes were saved, and the editor was closed.
+
+Then, the SSH service was restarted to apply the changes:
+
+```bash
+sudo systemctl restart sshd
+```
+
+## Webserver installation and configuration 
+
+### 1. Updated and Upgraded the Packages
+Before installing the web server, the system was updated by running the following commands:
+
+```bash
+sudo apt update
+sudo apt upgrade
+```
+
+### 2. Installed Nginx
+Next, Nginx was installed, which was used as the web server:
+
+```bash
+sudo apt install nginx
+```
```

Comment:
Better do not write technical documentation in a past tense - rather use present tense -> Next, install nginx on the server with the following command:
...